### PR TITLE
as updated patch route and prepares for requestor post route

### LINKED ISSUE #21: Added Patch Request Logic to Front End - Requestor

### PR DESCRIPTION
**Pull request by Gear-Smith**
closes #21  Completed the patch request logic, no mater that it looks terrible. Prepares requester for post request to create a 9-line item in the data base.

### CODE REVIEW

File: server/src/main/java/spring/ItemController.java [hunk 1]
```diff
@@ -43,8 +43,32 @@ public Item createItem(@RequestBody Item item) {
   public Item updateItem(@PathVariable Integer id, @RequestBody Map<String, String> body) throws Exception {
     Item item = itemRepo.findById(id).orElseThrow();
 
-    // if (body.get("key") != null)
-    // else throw new InvalidRequestException();
+    body.forEach((key, value) -> {
```
Comment (adamhl8) [inline server/src/main/java/spring/ItemController.java:46]: I think we can do something like this to patch fields regardless of what they are.

```
BeanWrapper accessor = PropertyAccessorFactory.forBeanPropertyAccess(item);
Field[] itemFields= item.getClass().getDeclaredFields();
for (Field field : itemFields) {
  String fieldName = field.getName();
  if (body.containsKey(fieldName)) accessor.setPropertyValue(fieldName, body.get(fieldName));
}
```

**Review by adamhl8 — CHANGES_REQUESTED** [review]
Looks good! Commented a potential solution for a more generic patch endpoint.

File: server/src/main/java/spring/ItemController.java (hunk 1 above)
Comment (Gear-Smith) [reply server/src/main/java/spring/ItemController.java]: removed my Alabama style logic and added the dank version.

**Review by adamhl8 — APPROVED** [review]
👍

**Comment by Gear-Smith** [comment]
CRAAAM

**Pull request merged**